src/calculation/ClauseResultsBuilder.ts

```diff
@@ -363,100 +372,215 @@ export function prettyResult(result: any | null, indentLevel?: number, keyIndent
 }
 
+/**
+ * Uses the pieces of a code object to show code, system, and display in a pretty way
+ * for whichever properties are available.
+ */
+export function prettyCode(
+  code: string,
+  system: string | undefined,
+  display: string | undefined,
+  includeType: boolean
+) {
+  const typeStr = includeType ? 'CODE: ' : '';
+  const displayStr = display ? `, ${display}` : '';
+  let systemStr = 'UNDEFINED_SYSTEM';
+  if (system) {
+    if (system in systemMap) {
+      systemStr = systemMap[system];
+    } else {
+      systemStr = system;
+    }
+  }
+  return `${typeStr}${systemStr} ${code}${displayStr}`;
+}
+
+/**
+ * Uses the pieces of a codeable concept object to show concept display text (if available)
+ * and the contained codings that belong to the concept in a pretty way
+ */
+export function prettyConcept(
+  conceptTxt: string | undefined,
+  conceptCodings: any[] | undefined,
+  includeType: boolean,
+  indentLevel?: number,
+  keyIndent?: number
+) {
+  let prettyConcept = '';
+  let keyDiff = 0;
+  if (includeType) {
+    prettyConcept = 'CONCEPT: ';
+    keyDiff = 2; //offset between concept key and code key including bracket
+  }
+
+  let conceptTxtNL = '';
+  if (conceptTxt || includeType) {
+    // include new line (+ indentation) if there's something to help dilineate the coding list
+    const keyIndentation = Array((keyIndent ?? 0) + keyDiff).join(' ');
+    conceptTxtNL = `\n${keyIndentation}`;
+  }
+
+  if (conceptTxt) {
+    prettyConcept = prettyConcept.concat(`${conceptTxt}`);
+  }
+  if (conceptCodings) {
+    prettyConcept = prettyConcept.concat(
+      `${conceptTxtNL}${prettyResult(
+        conceptCodings,
+        includeType,
+        indentLevel,
+        (keyIndent ?? 0) + (includeType ? keyDiff + 1 : keyDiff) // +1 for bracket
+      )}`
+    );
+  }
+  return prettyConcept;
+}
+
+/**
+ * Function to help prettyResult handle FHIRObject(s) specifically, handling similar types, but
+ * in a FHIRObject structure rather than a CQL object structure. Uses similar inputs put expects
+ * a FHIRObject result to be passed in.
+ */
+export function prettyFHIRObject(
+  result: FHIRObject,
+  includeType = true,
+  indentLevel?: number,
+  keyIndent?: number
+): string {
+  const resourceType = result.getTypeInfo()._name;
+  const keyIndentation = Array(keyIndent).join(' ');
+  const typeStr = includeType ? `${resourceType.toUpperCase()}: ` : '';
+  if (resourceType === 'Period') {
+    return `${typeStr}${prettyResult(result.get('start'))} - ${prettyResult(result.get('end'))}`;
+  } else if (resourceType === 'dateTime') {
+    return `${moment.utc(result.get('value').toString()).format('MM/DD/YYYY h:mm A')}`;
+  } else if (resourceType === 'date') {
+    return `${moment.utc(result.get('value').toString()).format('MM/DD/YYYY')}`;
+  } else if (resourceType === 'Coding') {
+    return prettyCode(
+      result.get('code')?.['value'],
+      result.get('system')?.['value'],
+      result.get('display')?.['value'],
```

Comment:
```suggestion
      result.get('code')?.value,
      result.get('system')?.value,
      result.get('display')?.value,
```

Reply:
oh - I didn't realize this would work since it doesn't work one level up, but this is much better!